docs/src/lang/records.md

```diff
@@ -148,17 +145,18 @@ is picked with `\E r \in [ field_1: S_1, ..., field_n: S_n]`.
 **Example in TLA+:**
 
 ```tla
-  [ name: { "A", "B", "C" }, port: 1..65535 ]
+  [ name: { "A", "B", "C" }, port: 65533..65535 ]
     \* A set of records. Each has two fields:
     \* field "name" that has the value from the set { "A", "B", "C" }, and
-    \* field "port" that has the value from the set 1..65535.
+    \* field "port" that has the value from the set 65533..65535.
 ```
 
-**Example in Python:** TLA+ functions are immutable, so we are using [frozendict]:
+**Example in Python:** We use a list instead of a set, since Python dictionaries are not hashable.
 
 ```python
-  frozenset({ frozendict({ "name": n, "port": p })
-                for n in { "A", "B", "C" } for p in range(1, 65535 + 1) })
+>>> [ { "name": n, "port": p } for n in [ "A", "B", "C" ] for p in range(65533, 65535 + 1) ]
+[{'name': 'A', 'port': 65533}, {'name': 'A', 'port': 65534}, {'name': 'A', 'port': 65535}, {'name': 'B', 'port': 65533}, {'name': 'B', 'port': 65534}, {'name': 'B', 'port': 65535}, {'name': 'C', 'port': 65533}, {'name': 'C', 'port': 65534}, {'name': 'C', 'port': 65535}]
+
 ```
```

Comment:
Let's just use frozen dict where it's needed (as in this case). Where we can omit frozen dict for simplicity, it is worth the gains for clearer exposition, but we shouldn't use different, incorrect data structures to get around using the correct data structure.

Reply:
I think we should only use data structures which are part of the language. frozendict is an external package which does not work with recent Python versions.

Reply:
I should have put this information in the PR message, it was a bit hidden in the commit message. Maybe there are ways to get a more faithful analogy within standard Python but personally I don't see the need to make this more complicated. There is surely some other package which could be used instead of frozendict, but I feel strongly that this should be avoided.

Reply:
I agree with igor's remarks here: https://github.com/informalsystems/apalache/pull/742#issuecomment-822511588

If you'd want to change the use of dictionaries to instead use namedtuples consistently, and add the additional explanatory passages needed to help motivate and explicate this for readers, we'd accept that in this PR.

Otherwise, let's just stick with `frozendicts` here.

We all agree that using a totally different data structure (lists instead of sets) is not compatible with our aims for this documentation, and we don't mind the dependency on the frozendict library if it allows a more accurate encoding.

Let me know if you  have any questions!